apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationNotificationList.jsx

```diff
@@ -2,42 +2,73 @@ import React, { Component } from 'react'
 import django from 'django'
 
 import { ModerationNotification } from './ModerationNotification'
-import { FilterBar } from './FilterBar'
+import { Filter } from './Filter'
 import { alert as Alert } from 'adhocracy4'
 
+const pendingFilterItems = [
+  { label: django.pgettext('kosmo', 'Pending'), value: 'true' },
+  { label: django.pgettext('kosmo', 'Archived'), value: 'false' },
+  { label: django.pgettext('kosmo', 'View all'), value: '' }
+]
+
+const classificationFilterItems = [
+  { label: django.pgettext('kosmo', 'Offensive'), value: 'OFFENSIVE' },
+  { label: django.pgettext('kosmo', 'Engaging'), value: 'ENGAGING' },
+  { label: django.pgettext('kosmo', 'Fact claiming'), value: 'FACTCLAIMING' },
+  { label: django.pgettext('kosmo', 'All categories'), value: '' }
+]
+
 export default class ModerationNotificationList extends Component {
   constructor (props) {
     super(props)
 
     this.state = {
       moderationComments: [],
-      filterItem: { filter: '', name: '' },
+      selectedFilters: { pending: '', classification: '' },
       isLoaded: false,
       alert: undefined
     }
   }
 
   componentDidMount () {
-    this.loadData(this.state.filterItem.filter)
+    this.loadData()
     setInterval(
-      () => !this.timer && this.loadData(this.state.filterItem.filter),
+      () => !this.timer && this.loadData(),
       3000
     )
   }
 
-  filterChangeHandle (filterItem) {
+  pendingFilterChangeHandle (value) {
     this.setState({
-      filterItem: filterItem,
+      selectedFilters: {
+        ...this.state.selectedFilters,
+        pending: value
+      },
       isLoaded: false
-    })
-    this.loadData(filterItem.filter)
+    },
+    () => this.loadData()
```

Comment:
Sorry, missing React-knowledge: could you explain what the `...` and `() =>` does?

Reply:
@fuzzylogic2000 Sure! The ... is the spread operator, same as ** in python. It unpacks the key/values of the dictionary, so here it means, that it takes all entries from this.state.selectedFilters.

The () => is function notation, I needed that, because I use loadData() here as a callback to setState (so it only loads the date once the state is updated), which expects a function. But since loadData itself is async, it returns a promise and thus I needed to wrap it again in a function (also found that by try and error).

Hope that makes sense?